spec.bs

```diff
@@ -1056,35 +1060,33 @@ Issue: Finish specifying the following algorithms.
 </div>
 
 <div algorithm>
-  The <dfn method for="InterestGroupBiddingScriptRunnerGlobalScope">setBid(|generateBidOutput|)</dfn>
-  method steps are:
 
-  1. Clear the bid. TODO: expand.
+To <dfn>convert GenerateBidOutput to generated bid</dfn> given a {{GenerateBidOutput}} |generateBidOutput|:
   1. If |generateBidOutput|["{{GenerateBidOutput/bid}}"] is less than or equal to 0, return true.
   1. If it's a component auction (TODO: how to integrate with runAdAuction), but
-    |generateBidOutput|["{{GenerateBidOutput/allowComponentAuction}}"] is false, return false.
-  1. Let |bidToSet| be a new [=generated bid=].
-  1. Set |bidToSet|'s [=generated bid/bid=] to |generateBidOutput|["{{GenerateBidOutput/bid}}"].
+    |generateBidOutput|["{{GenerateBidOutput/allowComponentAuction}}"] is false, return failure.
+  1. Let |bid| be a new [=generated bid=].
+  1. Set |bid|'s [=generated bid/bid=] to |generateBidOutput|["{{GenerateBidOutput/bid}}"].
   1. If |generateBidOutput|["{{GenerateBidOutput/ad}}"] [=map/exists=]:
     1. Let |adJson| be the result of [=serializing a JavaScript value to a JSON string=], given
       |generateBidOutput|["{{GenerateBidOutput/ad}}"].
     1. If |adJson| is failure, return failure.
-    1. Set |bidToSet|'s [=generated bid/ad=] to |adJson|.
+    1. Set |bid|'s [=generated bid/ad=] to |adJson|.
   1. Let |adDescriptor| be a new [=ad descriptor=].
   1. If |generateBidOutput|["{{GenerateBidOutput/adRender}}"] is a {{DOMString}}:
     1. Let |adUrl| be the result of running the [=URL parser=] on
       |generateBidOutput|["{{GenerateBidOutput/adRender}}"].
-    1. If |adUrl| is an error, return false.
-    1. If [=validating an ad url=] given |adUrl| returns false, return false.
+    1. If |adUrl| is an error, return failure.

```

Comment:
See See https://github.com/WICG/turtledove/issues/542#issuecomment-1509004711 - we might as well re-use the dfn we have, rather than invoke the URL parser directly (here and elsewhere).

Reply:
Done, although this one is genuinely a URL, not an origin.